test/generator/CMakeLists.txt

```diff
@@ -0,0 +1,301 @@
+        set(DEPS "external_code_generator_deps")
+    endif ()
+
+    # Most of our Generators are one-per-file...
+    set(NAMES "${NAME}")
+    # ...but some have multiple-per-file
+    if ("${NAME}" STREQUAL "nested_externs")
+        set(NAMES nested_externs_root nested_externs_inner nested_externs_combine nested_externs_leaf)
+    endif ()
+    foreach (N ${NAMES})
+        halide_generator("${N}.generator"
+                GENERATOR_NAME "${N}"
+                SRCS "${GEN_TEST_DIR}/${NAME}_generator.cpp"
+                DEPS ${DEPS})
+    endforeach ()
+endforeach ()
+
+# Define a nontrivial depedency for external_code.generator
+set(EC32 "${PROJECT_BINARY_DIR}/${PROJECT_NAME}.build/external_code_extern_bitcode_32")
+add_custom_command(OUTPUT "${EC32}.bc"
+        DEPENDS "${GEN_TEST_DIR}/external_code_extern.cpp"
+        COMMAND ${CLANG} ${CXX_WARNING_FLAGS} -O3 -c -m32 -target le32-unknown-nacl-unknown -emit-llvm "${GEN_TEST_DIR}/external_code_extern.cpp" -o "${EC32}.bc")
+add_custom_command(OUTPUT "${EC32}.cpp"
+        DEPENDS "${EC32}.bc"
+        COMMAND binary2cpp external_code_extern_bitcode_32 < "${EC32}.bc" > "${EC32}.cpp")
+
+set(EC64 "${PROJECT_BINARY_DIR}/${PROJECT_NAME}.build/external_code_extern_bitcode_64")
+add_custom_command(OUTPUT "${EC64}.bc"
+        DEPENDS "${GEN_TEST_DIR}/external_code_extern.cpp"
+        COMMAND ${CLANG} ${CXX_WARNING_FLAGS} -O3 -c -m64 -target le64-unknown-unknown-unknown -emit-llvm "${GEN_TEST_DIR}/external_code_extern.cpp" -o "${EC64}.bc")
+add_custom_command(OUTPUT "${EC64}.cpp"
+        DEPENDS "${EC64}.bc"
+        COMMAND binary2cpp external_code_extern_bitcode_64 < "${EC64}.bc" > "${EC64}.cpp")
+
+set(ECCPP "${PROJECT_BINARY_DIR}/${PROJECT_NAME}.build/external_code_extern_cpp_source")
+add_custom_command(OUTPUT "${ECCPP}.cpp"
+        DEPENDS "${GEN_TEST_DIR}/external_code_extern.cpp"
+        COMMAND binary2cpp external_code_extern_cpp_source < "${GEN_TEST_DIR}/external_code_extern.cpp" > "${ECCPP}.cpp")
+
+add_library(external_code_generator_deps "${EC32}.cpp" "${EC64}.cpp" "${ECCPP}.cpp")
+
+# ------ Generator tests for just-in-time mode: ------
+# Find all the files of form "foo_jittest.cpp" in test/generator
+# and declare a rule for them, each of which depends on foo.generator.
+file(GLOB JITTEST_SRCS RELATIVE "${GEN_TEST_DIR}" "${GEN_TEST_DIR}/*_jittest.cpp")
+foreach (FILE ${JITTEST_SRCS})
+    string(REPLACE "_jittest.cpp" "" NAME "${FILE}")
+    set(TARGET "generator_jit_${NAME}")
+    halide_project("${TARGET}" "generator" "${GEN_TEST_DIR}/${FILE}")
+    target_link_libraries("${TARGET}" PRIVATE "${NAME}.generator")
+    add_halide_test("${TARGET}"
+            GROUPS generator travis)
+endforeach ()
+
+# ------ Generator tests for ahead-of-time mode: ------
+# Find all the files of form "foo_aottest.cpp" in test/generator
+# and declare a rule for them, each of which depends on foo.generator.
+# Note that there are many special cases.
+# TODO(srj)
```

Comment:
...is this still a WIP?

Reply:
Unclear. I haven't really touched the generator files. To do so would require me to first update Halide's cmake rules (halide.cmake).

Reply:
Gotcha. (Obviously we can't land this until we're sure the Generator rules and tests are good to go.)

Reply:
I might not have been clear -- the generator rules and tests have already run successfully on Travis. I just haven't modified the build rules much. that TODO existed when I started on this project.